src/kernel.ts

```diff
@@ -900,9 +887,9 @@ class KernelFutureHandler extends Disposable implements IKernelFuture {
         }
       }
     } else if (msg.channel === 'shell') {
-      var reply = this._output;
+      var reply = this._reply;
```

Comment:
This was also a bug. The reply handler should be invoked here, not the output handler.

The test suite should be updated to exercise this code.